spec/features/upload_descriptive_metadata_spec.rb

```diff
@@ -13,32 +13,65 @@
     FactoryBot.create_for_repository(:persisted_item)
   end
 
-  let(:csv) do
-    "title1.value,purl\nmy title,https://purl.stanford.edu/#{Druid.new(item.externalIdentifier).without_namespace}\n"
-  end
-  let(:file) do
-    Tempfile.new(%w[upload .csv]).tap do |file|
-      file.write csv
-      file.close
-    end
+  before do
+    sign_in user, groups: ['sdr:administrator-role']
+    visit solr_document_path(item.externalIdentifier)
   end
 
   after do
     file.unlink
   end
 
-  before do
-    sign_in user, groups: ['sdr:administrator-role']
-    visit solr_document_path(item.externalIdentifier)
+  context 'with a csv file' do
+    let(:csv) do
+      "title1.value,purl\nmy title,https://purl.stanford.edu/#{Druid.new(item.externalIdentifier).without_namespace}\n"
+    end
+    let(:file) do
+      Tempfile.new(%w[upload .csv]).tap do |file|
+        file.write csv
+        file.close
+      end
+    end
+
+    it 'uploads csv descriptive' do
+      click_button 'Manage description'
+      click_link 'Upload Cocina spreadsheet'
+
+      attach_file("Upload Cocina descriptive metadata spreadsheet for #{item.externalIdentifier}", file.path)
+      click_button 'Upload'
+
+      expect(page).to have_content 'Descriptive metadata has been updated.'
+    end
   end
 
-  it 'uploads descriptive' do
-    click_button 'Manage description'
-    click_link 'Upload Cocina spreadsheet'
+  context 'with an excel file' do
+    let(:file) do
+      Tempfile.new(%w[upload .xlsx]).tap do |file|
+        # Open the temp file as an XLSX workbook
+        workbook = WriteXLSX.new(file.path)
```

Comment:
Instead of adding a new dependency, can you just add a sample xls in `spec/fixtures/files` ?

Reply:
No - it requires a matching purl, so we need to be able to write it at the time of the test the same way we do with the CSV.

Reply:
Okay.